src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,241 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association;
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association.Rate;
+
+/**
+ * Aggregation model for State aggregation. Only contains info useful for
+ * analysis.
+ */
+public class AggregatedState {
+
+	public static class AggregatedRate {
+		public long bitRate;
+		public int chWidth;
+		public List<Integer> mcs;
+
+		/** Constructor with no args */
+		public AggregatedRate() {
+			mcs = new ArrayList<>();
+		}
+
+		/** Add a Rate to the AggregatedRate */
+		public void add(Rate rate) {
+			if (rate == null) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				bitRate = rate.bitrate;
+				chWidth = rate.chwidth;
+				mcs = new ArrayList<>();
+			}
+			mcs.add(rate.mcs);
+		}
+
+		/**
+		 * Add an AggregatedRate with the same channel_width to the
+		 * AggregatedRate
+		 */
+		public void add(AggregatedRate rate) {
+			if (rate.chWidth != chWidth) {
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				bitRate = rate.bitRate;
+				chWidth = rate.chWidth;
+			}
+			mcs.addAll(rate.mcs);
+		}
+	}
+
+	public static class Radio {
+		public int channel;
+		public int channelWidth;
+		public int txPower;
+
+		public Radio() {}
+
+		public Radio(int channel, int channelWidth, int txPower) {
+			this.channel = channel;
+			this.channelWidth = channelWidth;
+			this.txPower = txPower;
+		}
+
+		public Radio(Map<String, Integer> radioInfo) {
+			channel = radioInfo.getOrDefault("channel", -1);
+			channelWidth = radioInfo.getOrDefault("channel_width", -1);
+			txPower = radioInfo.getOrDefault("tx_power", -1);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(channel, channelWidth, txPower);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+
+			Radio other = (Radio) obj;
+			return channel == other.channel &&
+				channelWidth == other.channelWidth && txPower == other.txPower;
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rxBytes;
+	public long rxPackets;
+	public AggregatedRate rxRate;
+	public long txBytes;
+	public long txDuration;
+	public long txFailed;
+	public long txPackets;
+	public AggregatedRate txRate;
+	public long txRetries;
+	public int ackSignal;
+	public int ackSignalAvg;
+	public Radio radio;
+
+	/** Constructor with no args */
+	public AggregatedState() {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		this.radio = new Radio();
+	}
+
+	/** Construct from Aggregatedstate */
+	public AggregatedState(AggregatedState state) {
```

Comment:
Is this constructor used? If not, delete.

If you keep it:
- maybe you can do just `this()` followed by `add(state)`? I think it would be "simpler" (less duplicate code, fewer lines), but the downside is the radio field would get written then overwritten
- change the javadoc's `Aggregatedstate` -> `AggregatedState` (capitalized `S`)